### ISSUE TITLE
KD Compression

### ISSUE THREAD

**Issue by gjairath** [chumingqian/Model_Compression_For_YOLOV3-V4#5]
Hey,

I was using your repo to do KD on Yolov4, I had a question: do teacher and student share same original cfg file for Yolov4? Or is the teacher model structurally pruned first? This is a low level question I can't find an answer to.  Can I just skip the --t_cfg parameter and use the default option instead? Is it the same for the --t_weights? I'm using pretrained weights from onnx, both teacher student share the same weights?

Thanks, 
gjairath

**Comment by chumingqian** [comment]
Hi, @gjairath :
            Usually  the  teacher and student will not share the same cfg or weights.  
1. In the knowledge distillation way:  The original cfg  and it's corresponding  weight ( unpruned model ) will be as the teacher network,    and   use the pruned model as the  student network .

2.  After we used pruned way for the  original cfg and weights,   we will  get a new cfg file  and a  new weights,  usually it's weights  smaller  then the origianl weights, but  it's precision will  also decline.  ( Before we pruned the model,  we neeed to sparse training for the original model to select out unimportant channel,  then we pruned these unimportant channels)

3.  In the step2,  we get a smaller model and treat it  as the student network.   So  we use  knowledge distillation to  improve the  
student network's(the pruned network)  accuracy. 

You are welcome.

**Comment by gjairath** [comment]
I understand, thank you. 

Would 3-5 epochs suffice sparse training, and max possible percent for channel pruning?

Do you suggest any reading for such lower level questions?

Thanks again.
gjairath

**Comment by chumingqian** [comment]
Hi,  gjairth:

1. Usually 100 - 200 epochs(  it depends on your network's size,  if  your  network  is deep like more then 200 layers)   for sparse training.   

2. The max possible percent depends on your  sparse training effect  and your  task;
 If you sparse effect is good (most of  the weight  distibution value will close to  zero,) 
   and  your task is simple enough ( your dataset : 1 class?  VOC - 20 classes?  COCO -80 classes?)   you can set a high prune percent ( like 0.6- 0.8 ).


3. This paper i recommonded : Learning Efficient Convolutional Networks through Network Slimming  website:  https://arxiv.org/abs/1708.06519

4.  Please  do not  think  your  questions  is low  level,   you can do more and read more ,and you will become better then  now.

My pleasure.
